lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3269,32 +3269,52 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
 } // namespace
 
 namespace {
-class ConvertAtenOnesOp : public OpConversionPattern<AtenOnesOp> {
-public:
-  using OpConversionPattern::OpConversionPattern;
+// Converts AtenOnesOp and AtenZerosOp.
+struct ConvertAtenOnesZerosOp : ConversionPattern {
+  ConvertAtenOnesZerosOp(TypeConverter &typeConverter, MLIRContext *context)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag(), /*benefit=*/1,
+                          context) {}
+
   LogicalResult
-  matchAndRewrite(AtenOnesOp op, OpAdaptor adaptor,
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
+    if (!isa<AtenOnesOp, AtenZerosOp>(op))
+      return rewriter.notifyMatchFailure(op,
+                                         "not a supported ones or zeros op");
+
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
-    Location loc = op.getLoc();
+    Location loc = op->getLoc();
+
+    SmallVector<Value, 3> opArguments;
+    int64_t elementValue;
+
+    if (AtenOnesOp onesOp = dyn_cast<AtenOnesOp>(op)) {
+      opArguments.insert(opArguments.end(),
```

Comment:
nit: It might make things clearer to use `opArguments = {...};` (or if that doesn't work, you can do `opArguments.assign({...});` The argument `opArguments.end()` makes it seem as if there could be other arguments already in the vector.